src/server/search/search_family.cc

```diff
@@ -110,6 +117,47 @@ optional<SearchParams> ParseSearchParamsOrReply(CmdArgList args, ConnectionConte
       continue;
     }
 
+    // RETURN {num} [{ident} AS {name}...]
+    if (ArgS(args, i) == "RETURN") {
+      if (i + 1 >= args.size()) {
+        (*cntx)->SendError(kSyntaxErr);
+        return nullopt;
+      }
+
+      uint64_t num_fields = args.size();
+      if (!absl::SimpleAtoi(ArgS(args, i + 1), &num_fields)) {
+        (*cntx)->SendError(kSyntaxErr);
+        return nullopt;
+      }
+
+      i += 1;
```

Comment:
Consider moving this on line 122 such that we don't use the `i + 1` everywhere.

Reply:
I don't wanna polish this function given that I'll tear it down completely